src/SQLProvider.Runtime/Providers.MySql.fs

```diff
@@ -385,14 +385,26 @@ type internal MySqlProvider(resolutionPath, contextSchemaPath, owner:string, ref
             |> fun (x,_)-> x
             |> List.rev
             |> List.toArray
-            |> Array.unzip
+
+        let columnNames, values = columnNamesWithValues |> Array.unzip
 
         sb.Clear() |> ignore
-        ~~(sprintf "INSERT INTO %s (%s) VALUES (%s); SELECT LAST_INSERT_ID();"
+        ~~(sprintf "INSERT INTO %s (%s) VALUES (%s)"
             (entity.Table |> quotedTableName)
             ("`" + (String.Join("`, `",columnNames)) + "`")
             (String.Join(",",values |> Array.map(fun p -> p.ParameterName))))
 
+        match entity.OnConflict with
+        | Throw -> ()
+        | Update ->
+          ~~(sprintf " ON DUPLICATE KEY UPDATE %s"
+                (String.Join(",", columnNamesWithValues |> List.map(fun (c,p) -> sprintf "`%s`=%s" c p.ParameterName))))
+        | DoNothing ->
+          ~~(sprintf " AS x ON DUPLICATE KEY UPDATE %s"
```

Comment:
I think this `AS x` syntax might only be supported since MySQL 8. We could instead use the `VALUES()` syntax, but that is now deprecated. I have written this such that it would only be an issue if a user was to specify `OnConflict.DoNothing` as the `AS x` part is conditional on that. So this shouldn't regress any existing code given that `OnConflict` is currently not supported for `MySQL` anyway. 

If there is a way to determine the MySQL version then I could use that to generate different queries based on the version, but for now this seemed like the most pragmatic approach.

Reply:
The negative side of detecting version is that it's an extra network call to the db.

Reply:
Yes, I think I'll leave it as this then and it will just not be supported on older versions of MySQL as it currently isn't.

Reply:
My tests revealed that this was actually doing the wrong thing anyway, the `AS x` syntax is a way to refer to the values being inserted, not the existing column values as I had intended. So the correct implementation is simply to do `ON DUPLICATE KEY UPDATE col1 = col1, col2 = col2` etc which I believe is supported on older versions of MySQL too. 👍